src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -32,19 +33,27 @@ internal static IEnumerable<PropertyInfo> ExcludeProperties<TSourceType>(IEnumer
                 ? properties
                 : properties.Where(propertyInfo => !excludedProperties!.Any(p => GetPropertyName(p) == propertyInfo.Name));
 
-        internal static FieldType CreateField(PropertyInfo propertyInfo, Type graphType, bool isInputType)
+        /// <summary>
+        /// Creates a <see cref="FieldType"/> for the specified <see cref="MemberInfo"/>.
+        /// </summary>
+        internal static FieldType CreateField(MemberInfo memberInfo, Type graphType, bool isInputType)
         {
             var fieldType = new FieldType()
             {
-                Name = propertyInfo.Name,
-                Description = propertyInfo.Description(),
-                DeprecationReason = propertyInfo.ObsoleteMessage(),
+                Name = memberInfo.Name,
+                Description = memberInfo.Description(),
+                DeprecationReason = memberInfo.ObsoleteMessage(),
                 Type = graphType,
-                DefaultValue = isInputType ? propertyInfo.DefaultValue() : null,
+                DefaultValue = isInputType ? memberInfo.DefaultValue() : null,
+                Resolver = isInputType ? null : MemberResolver.Create(memberInfo),
             };
+            if (isInputType)
```

Comment:
Why only fro inputs?

Reply:
It is only used for ToObject within ParseDictionary. I can apply it to output graphs also but it is not necessary; the field resolver doesn’t need this metadata set to work properly. 

Once I add an overload for ParseDictionary then it won’t be needed at all. 

Should I just set it for both inputs and outputs anyway?

Reply:
I was also thinking it would be more valuable to store the MemberInfo in the metadata rather than just the field name. That way we don’t have to search the class for a matching property name so parsing variables should be faster.

Reply:
OK. Do as you see fit.